__tests__/app.test.js

```diff
@@ -0,0 +1,50 @@
+const request = require('supertest');
+const db = require('../db/connection');
+const app = require('../app');
+const seed = require('../db/seeds/seed');
+const { articleData, commentData, topicData, userData} = require('../db/data/test-data/index');
+
+afterAll(() => {
+  if (db.end) db.end();
+});
+  
+beforeEach(() => seed({articleData, commentData, topicData, userData}));
+
+describe("1) 404 : invalid path", () => {
+it("Responds with 404 if client makes a request to an invalid path", () => {
+return request(app)
+.get('/api/topisc')
+.expect(404)
+.then(({ body : { message }}) => {
+  expect(message).toBe('Invalid path');
+})
+})
+})
+
+describe("2) GET api/topics", () => {
+it("Responds with 200 status and an array of topic objects each with two keys: slug and description", () => {
+return request(app)
+.get('/api/topics')
+.expect(200)
+.then(({ body : { topics } }) => {
+expect(topics).toBeInstanceOf(Array);
+expect(topics).toHaveLength(3);
+expect(topics).toEqual([
```

Comment:
Whilst this assertion on line 32 isn't wrong as the test data is small and topics data is manageable to check this way what would be better is to get used to checking the 'shape' of the data when you're requesting multiple resources back. 

This means checking each item in topics has the right structure and data types and not necessarily the exact values.
I'd like you to change the assertion on line 32 to reflect this so we're getting into this pattern early on.

(also i like the assertion on 31)